internal/verifier/verifier.go

```diff
@@ -70,14 +61,23 @@ const (
 
 // Verifier is the object that verifies artifacts
 type Verifier struct {
-	verifier ArtifactVerifier
+	verifier verifyif.ArtifactVerifier
 	cacheDir string
 }
 
+// NewCustomVerifier creates a new Verifier object for testing using a custom verifier instance
+func NewCustomVerifier(verifier verifyif.ArtifactVerifier) *Verifier {
+	return &Verifier{
+		verifier: verifier,
+		cacheDir: "",
+	}
+}
+
 // NewVerifier creates a new Verifier object
 func NewVerifier(verifier Type, verifierURL string, containerAuth ...container.AuthMethod) (*Verifier, error) {
```

Comment:
Should this return a `verifyif.ArtifactVerifier`?

Reply:
for this we'd move the cache cleanup to be a method of that interface. It would make testing and mocking easier and get rid of the `NewCustomVerifier` flake. The original code was written by @rdimitrov and I'm not sure if he had any specific reason to keep the cache one level up from the verifier (the cache prefix is hardcoded to be sigstore but the switch-case in NewVerifier has sigstore only as one of the choices..)

Reply:
Yeah, I'd tend to prefer to put the Cache cleanup method in the interface, then have the constructors return the interface type.  As you point out, it makes replacing them with test fakes easier.